src/DIRAC/Resources/Catalog/FileCatalog.py

```diff
@@ -427,7 +427,7 @@ def _getCatalogs(self):
         return res
       catalogConfig = res['Value']
       if catalogConfig['Status'] == 'Active':
-        res = self._generateCatalogObject(catalogName)
+        res = self._generateCatalogObject(catalogName, **catalogConfig)
```

Comment:
Why do you need the `**`? Why not pass catalogConfig as an optionsDict dictionary parameter?

Reply:
By doing this, I don't have to test for the options being None in the line below:https://github.com/DIRACGrid/DIRAC/blob/6778bbef2a67f5d946bdb6cd1ab0261ec5034eb1/src/DIRAC/Resources/Catalog/FileCatalogFactory.py#L35

Besides I enforce it to be a dict.

Reply:
I think it would be better to be explicit that these functions now takes in addition a parameter that is a dictionary.

Reply:
Optionally, only if there are any options in Operations. Adding an '{}` as a default parameter is out of question, so I find ** an elegant way.

Reply:
It is elegant until someone adds another keyword argument, someone makes a typo somewhere and hilarity ensues because `**options`` swallows everything.

So then having `options=None` as a keyword argument, and adding two lines of checking for None seem like a small price to pay to me for having pylint save people from typos

Reply:
OK, stylistic choice, I can make a change you are suggesting. Another issue is that @chaen questions the usefulness of this PR ...